pepper-apis/src/main/resources/db-changes/schema/DDP-5573-activity-nesting.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet author="yufeng" id="20210212-activity-nesting-table">
+        <createTable tableName="activity_nesting">
+            <column name="activity_nesting_id" type="bigint" autoIncrement="true" startWith="1">
+                <constraints primaryKey="true" primaryKeyName="activity_nesting_pk"/>
+            </column>
+            <column name="parent_activity_id" type="bigint">
+                <constraints nullable="false"
+                             references="study_activity(study_activity_id)"
+                             foreignKeyName="activity_nesting_parent_activity_fk"/>
+            </column>
+            <column name="nested_activity_id" type="bigint">
+                <constraints nullable="false"
+                             references="study_activity(study_activity_id)"
+                             foreignKeyName="activity_nesting_nested_activity_fk"/>
+            </column>
+        </createTable>
+        <addUniqueConstraint tableName="activity_nesting"
+                             columnNames="parent_activity_id, nested_activity_id"
+                             constraintName="activity_nesting_parent_nested_uk"/>
+    </changeSet>
```

Comment:
Did you consider just adding a parent_activity_id column to study_activity?

Reply:
I did. However, I chose to go this route because:

1. It's more consistent with how we do "nesting" in other aspects. See tables for "nested blocks and "nested" composite questions.
2. It gives up room to grow. For example, composite question nesting table has `display_order` attribute. We don't need a "display order" for nested activities, but should we need to add more properties to describe parent/child activities, we have a place to do this with this `activity_nesting` table.

Reply:
I can't say I see the need for room for growth. Order does not appear to be relevant as we getting that from the containing blocks (sort of). The table does facilitate a many-to-many relationship, but the constraint added explicitly precludes it.
Not going to push the issue here. If you feel this is more consistent with the rest of the code and schema I will accept your judgement.

Reply:
After working with some real sql queries that deals with parent/child activities, I changed my mind and went the simpler solution of just adding new column. This simplifies the queries, and we can always pull it out into a new join table if needed.